api/controllers/MerchStoreController.ts

```diff
@@ -197,18 +204,57 @@ export class MerchStoreController {
     return originalOrder;
   }
 
-  @Patch('/order')
-  async editMerchOrder(@Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+  @Patch('/order/:uuid/pickup')
+  async editMerchOrderPickup(@Params() params: UuidParam,
+    @Body() editOrderRequest: EditMerchOrderPickupRequest,
     @AuthenticatedUser() user: UserModel): Promise<EditMerchOrderResponse> {
-    if (!PermissionsService.canFulfillMerchOrders(user)) throw new ForbiddenError();
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    await this.merchStoreService.editMerchOrderPickup(params.uuid, editOrderRequest.pickupEvent, user);
+    return { error: null };
+  }
+
+  @Post('/order/:uuid/cancel')
+  async cancelMerchOrder(@Params() params: UuidParam, @AuthenticatedUser() user: UserModel) {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const order = await this.merchStoreService.cancelMerchOrder(params.uuid, user);
+    return { error: null, order };
+  }
+
+  @Post('/order/:uuid/missed')
+  async markOrderAsMissed(@Params() params: UuidParam, @AuthenticatedUser() user: UserModel) {
+    if (!PermissionsService.canManageMerchOrders(user)) throw new ForbiddenError();
+    const order = await this.merchStoreService.markOrderAsMissed(params.uuid, user);
+    return { error: null, order };
+  }
+
+  @Post('/order/:uuid/fulfill')
+  async fulfillMerchOrderItems(@Params() params: UuidParam, @Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+    @AuthenticatedUser() user: UserModel): Promise<FulfillMerchOrderResponse> {
+    if (!PermissionsService.canManageMerchOrders(user)) throw new ForbiddenError();
     const numUniqueUuids = (new Set(fulfillOrderRequest.items.map((oi) => oi.uuid))).size;
     if (fulfillOrderRequest.items.length !== numUniqueUuids) {
       throw new BadRequestError('There are duplicate order items');
     }
-    await this.merchStoreService.updateOrderItems(fulfillOrderRequest.items);
+    await this.merchStoreService.fulfillOrderItems(fulfillOrderRequest.items, params.uuid, user);
     return { error: null };
   }
 
+  @Post('/order/cleanup')
+  async cancelAllPendingMerchOrders(@AuthenticatedUser() user: UserModel): Promise<CancelAllPendingOrdersResponse> {
```

Comment:
If this route isn't needed yet, can we make this PR a little smaller and put up another PR when it is needed?

Reply:
Just confirmed with Steven, we will be rolling with /order/cleanup route and removing the status query param - frontend can do any filtering if needed since there's no current use case for it